docs/core/baseapp.md

```diff
@@ -0,0 +1,331 @@
+# BaseApp
+
+## Pre-requisite Reading
+
+- [Anatomy of an SDK application](../basics/app-anatomy.md)
+- [Lifecycle of an SDK transaction](../basics/tx-lifecycle.md)
+
+## Synopsis
+
+This document describes `baseapp`, the abstraction that implements most of the common functionalities of an SDK application. 
+
+- [Introduction](#introduction)
+- [Type Definition](#type-definition)
+- [Constructor](#constructor)
+- [States](#states)
+- [Routing](#routing)
+- [Main ABCI Messages](#abci)
+    + [CheckTx](#checktx)
+    + [DeliverTx](#delivertx)
+- [RunTx, AnteHandler and RunMsgs](#runtx-,antehandler-and-runmsgs)
+    + [RunTx](#runtx)
+    + [AnteHandler](#antehandler)
+    + [RunMsgs](#runmsgs)
+- [Other ABCI Message](#other-abci-message)
+    + [InitChain](#initchain)
+    + [BeginBlock](#beginblock)
+    + [EndBlock](#endblock)
+    + [Commit](#commit)
+    + [Info](#info)
+    + [Query](#query)
+    
+## Introduction
+
+`baseapp` is a base class that implements the core of an SDK application, namely:
+
+- The [Application-Blockchain Interface](#abci), for the state-machine to communicate with the underlying consensus engine (e.g. Tendermint). 
+- A [Router](#routing), to route [messages](./tx-msgs.md) and [queries](./querier.md) to the appropriate [module](../building-modules/modules.md).
+- Different [states](#states), as the state-machine can have different parallel states updated based on the ABCI message received. 
+
+The goal of `baseapp` is to provide a boilerplate SDK application that developers can easily extend to build their own custom application. Usually, developers will create a custom type for their application, like so:
```

Comment:
Isn't the goal of `baseapp` to connect all the modules that make up an application and provide common utilities to them? I don't think it is to "provide boilerplate" IMO